torch/_functorch/partitioners.py

```diff
@@ -2646,6 +2646,11 @@ def classify_nodes(joint_module, static_lifetime_input_indices):
             )
     bw_module = reordering_to_mimic_autograd_engine(bw_module)
 
+    # raise all getitem ops to as early as possible
+    # this is helpful for memory, especially in the case of aot_eager backend
+    fw_module = raise_getitems(fw_module)
+    bw_module = raise_getitems(bw_module)
```

Comment:
Can you add a test? I would probably it add it in this file, similar to this test https://github.com/pytorch/pytorch/blob/main/test/functorch/test_aotdispatch.py#L3792, where you:

(1) Call the AOTAutograd API's on a function that exhibits this behavior (calls an aten op that returns a tuple, where one element of the tuple is used much later in the fn)

(2) use `self.assertExpectedInline()` to assert on the exact string forward graph, so we're asserting that the getitem calls were moved up properly